pkg/pattern/drain/line_tokenizer.go

```diff
@@ -38,7 +38,7 @@ func newPunctuationTokenizer() *punctuationTokenizer {
 }
 
 func (p *punctuationTokenizer) Tokenize(line string) ([]string, interface{}) {
-	tokens := make([]string, len(line))                  // Maximum size is every character is punctuation
+	tokens := make([]string, 0, 64)                      // Maximum size is every character is punctuation
```

Comment:
Was 64 just an educated guess? 
A potential follow up is to add a metric for "characters per token" or "tokens per line" so we can get a better estimate here - we can probably improve perf a bit more with a better start point.